src/reflect.js

```diff
@@ -34,7 +34,7 @@ if (typeof Reflect.metadata !== 'function') {
 
 if (typeof Reflect.defineProperty !== 'function') {
   Reflect.defineProperty = function(target, propertyKey, descriptor) {
-    if (typeof target !== 'object') {
+    if (typeof target === 'object' ? target === null : typeof target !== 'function') {
```

Comment:
See table 35 on http://www.ecma-international.org/ecma-262/6.0/#sec-typeof-operator. So it actually does make sense.
`Reflect.defineProperty` also throws a TypeError when the descriptor is not an object. Shall I update this PR accordingly?